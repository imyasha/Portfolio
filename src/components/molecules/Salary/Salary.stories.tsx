import { Salary, SalaryProps } from './Salary';

export default {
	title: 'Molecules/Salary',
	component: Salary,
};

export const SalaryStory = (args: SalaryProps) => <Salary {...args} />;

SalaryStory.storyName = 'Salary';
SalaryStory.args = {
	salaryRange: {
		minimum: 100000,
		median: 150000,
		maximum: 200000,
	},
};
