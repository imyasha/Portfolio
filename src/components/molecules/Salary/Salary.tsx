import { useEffect, useState } from 'react';

export interface SalaryProps {
	salaryRange: {
		minimum: number;
		median: number;
		maximum: number;
	};
}

const Salary = ({ salaryRange }: SalaryProps) => {
	const [selectedSalary, setSelectedSalary] = useState(
		salaryRange.minimum - 40_000
	);
	const [emoji, setEmoji] = useState('❓');

	const { minimum, median, maximum } = salaryRange;

	useEffect(() => {
		if (selectedSalary <= minimum - 30_000) {
			setEmoji('😭');
		} else if (
			selectedSalary > minimum - 30_000 &&
			selectedSalary <= minimum - 20_000
		) {
			setEmoji('😢');
		} else if (
			selectedSalary > minimum - 20_000 &&
			selectedSalary <= minimum - 10_000
		) {
			setEmoji('🙁');
		} else if (selectedSalary > minimum - 10_000 && selectedSalary < minimum) {
			setEmoji('😕');
		} else if (selectedSalary >= minimum && selectedSalary < minimum + 10_000) {
			setEmoji('😏');
		} else if (
			selectedSalary > minimum + 10_000 &&
			selectedSalary <= minimum + 20_000
		) {
			setEmoji('🙂');
		} else if (
			selectedSalary > minimum + 20_000 &&
			selectedSalary <= median + 20_000
		) {
			setEmoji('😀');
		} else if (selectedSalary >= maximum) {
			setEmoji('🤑');
		}
	}, [maximum, median, minimum, selectedSalary]);

	return (
		<div>
			<p>
				<strong>Salary Expectation: </strong>
				There are several factors that influence the salary expectation I have
				for a future role, but here&lsquo;s a good approximation (please
				don&lsquo;t make me cry):
			</p>
			<input
				type="range"
				min={minimum - 40_000}
				max={maximum + 50_000}
				value={selectedSalary}
				step={5_000}
				onChange={(e) => setSelectedSalary(Number(e.target.value))}
				className="mx-2"
				data-testid="salaryRange"
			/>
			<label htmlFor="salaryRange">
				{Intl.NumberFormat('en-GB', {
					style: 'currency',
					currency: 'GBP',
				}).format(selectedSalary)}
			</label>
			<p>
				<strong>Happiness Score:</strong>
				<span className="text-3xl"> {emoji}</span>
			</p>
		</div>
	);
};

export { Salary };
