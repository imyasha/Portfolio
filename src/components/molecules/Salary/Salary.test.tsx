import { Salary } from '../Salary';
import { fireEvent, render } from '@testing-library/react';

describe('Salary', () => {
	const testRange = {
		minimum: 100000,
		median: 150000,
		maximum: 200000,
	};

	it('renders correctly', () => {
		const { container } = render(<Salary salaryRange={testRange} />);
		expect(container).toMatchSnapshot();
	});
});
