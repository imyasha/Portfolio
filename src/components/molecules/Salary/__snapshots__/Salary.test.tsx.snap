// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Salary renders correctly 1`] = `
<div>
  <div>
    <p>
      <strong>
        Salary Expectation: 
      </strong>
      There are several factors that influence the salary expectation I have for a future role, but hereâ€˜s a good approximation (please donâ€˜t make me cry):
    </p>
    <input
      class="mx-2"
      data-testid="salaryRange"
      max="250000"
      min="60000"
      step="5000"
      type="range"
      value="60000"
    />
    <label
      for="salaryRange"
    >
      Â£60,000.00
    </label>
    <p>
      <strong>
        Happiness Score:
      </strong>
      <span
        class="text-3xl"
      >
         
        ðŸ˜­
      </span>
    </p>
  </div>
</div>
`;
